packages/aws-rfdk/lib/deadline/scripts/bash/configureWorker.sh

```diff
@@ -113,6 +117,13 @@ if [ ${#WORKER_POOLS[@]} -gt 0 ]; then
   "$DEADLINE_COMMAND" -SetPoolsForSlave $(IFS=, ; echo "${WORKER_NAMES[*]}") $(IFS=, ; echo "${WORKER_POOLS[*]}")
 fi
 
+# Setting listening port for the workers on this node (we cannot have multiple workers listening on the same port)
+port_offset=0
+for worker_name in "${WORKER_NAMES[@]}"; do
+  "$DEADLINE_COMMAND" -ExecuteScriptNoGui "$WORKER_LISTENING_PORT_CONFIG_SCRIPT" -n $worker_name -p $(($WORKER_LISTENING_PORT+$port_offset))
+  port_offset=$((port_offset+1))
```

Comment:
Looking back at this, I never added documentation to explain the port number would be incremented by one for any additional workers found on the host.

I'm not sure if it's better to add documentation for this, or just remove the feature. I think technically it's possible to support multiple workers on a host right now, but I believe in the future this won't work with auto-scaling.

Reply:
Hrm. This is tricky/complicated. The WorkerFleet doesn't currently know how many workers are running on the node, so it's not opening up ports with `port_offset > 0` in the security group.

We are also planning to be very opinionated about one worker per node for autoscaling, but maybe not everyone wants to use the autoscaling...

Some options, off the top of my head, are:
1. We assume no more than, say, 8 workers on a node and always open up a range of ports in the security group.
2. Have this only support single worker.
3. Some sort of way for the customer to tell us how many worker applications are running on the instance; though, that feels like it could balloon into needing to also configure that many workers to start.
4. other?

Thoughts?

Reply:
I know Josh has said there's a use case for multiple workers on a single host to share licenses, so if we can support it without too much work, I think we should. I like the idea of limiting it to 8 and opening up that range in the security group, that keeps it simple and we can always make improvements in the future if we find customers need more.